### PR TITLE
Switch DisableNodeKubeProxyVersion back to disabled-by-default

### PR DESCRIPTION
**Pull request by liggitt**
This is clearing a stable API field, so the 1 year from announcement to change period applies

#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:

I missed the kubeProxy deprecation tried to proceed to default-enabled in just two releases. Because it is modifying a v1 API surface, it should give a year from announcement to change.

https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior

Reverts https://github.com/kubernetes/kubernetes/pull/123845

#### Does this PR introduce a user-facing change?

```release-note
Reverted the `DisableNodeKubeProxyVersion` feature gate to default-off to give a full year from deprecation announcement in 1.29 to clearing the field by default, per the [Kubernetes deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
```

Related to https://github.com/kubernetes/enhancements/issues/4004

### CODE REVIEW

**Comment by jpbetz** [comment]
/lgtm
/approve

Good catch

**Comment by sftim** [comment]
Changelog suggestion (scroll along to see it all)
```diff
-Reverted DisableNodeKubeProxyVersion feature gate to default-off to give a full year from deprecation announcement in 1.29 to clearing the field by default, per the Kubernetes deprecation policy at https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+Reverted the `DisableNodeKubeProxyVersion` feature gate to default-off to give a full year from deprecation announcement in 1.29 to clearing the field by default, per the [Kubernetes deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
```

**Comment by liggitt** [comment]
the linter is incorrect, this is describing the state of the feature, and is not a deprecation comment on the Go constant

**Comment by liggitt** [comment]
docs update in https://github.com/kubernetes/website/pull/47511/files

**Comment by liggitt** [comment]
is pull-kubernetes-verify-lint really blocking? the insistence on case of the deprecated comment is wrong

**Comment by liggitt** [comment]
repushed tweaking the case of the comment to make the linter happy :-/

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1057,7 +1056,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	DisableKubeletCloudCredentialProviders: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 
-	DisableNodeKubeProxyVersion: {Default: true, PreRelease: featuregate.Beta},
+	DisableNodeKubeProxyVersion: {Default: false, PreRelease: featuregate.Deprecated}, // default on in 1.33
```
Comment (HirazawaUi) [inline pkg/features/kube_features.go:1059]: Just wondering, why can't we follow the existing pattern in the KEP and return it to the alpha stage, the current approach will cause unit tests to fail and we need to modify the KEP according to it.

**Comment by BenTheElder** [comment]
> is pull-kubernetes-verify-lint really blocking? 

Yes, this is most of the go static analysis (anything current)

> the insistence on case of the deprecated comment is wrong

We should fix that. If we think the sub-linter is wrong it can be disabled or even just that specific check.

I looked upstream though, and go seems to use `Deprecated:` consistently, e.g. https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/io/ioutil/ioutil.go;l=89

**Comment by BenTheElder** [comment]
verify-featuregates failing in the main job is from https://github.com/kubernetes/kubernetes/pull/125830

File: pkg/features/kube_features.go (hunk 1 above)
Comment (liggitt) [reply pkg/features/kube_features.go:1059]: alpha is not the thing failing the unit tests, defaulting back to false is what's failing, and means we have to revert the unit test changes as well

pushed an update reverting the node unit test change

**Review by BenTheElder — COMMENTED** [review]
/lgtm

**Comment by thockin** [comment]
This is a GA field, but is  demonstrably unreliable, bordering on useless, but I guess this is proper.  Sorry for approving it in the first place.

/approve

File: pkg/features/kube_features.go (hunk 1 above)
Comment (HirazawaUi) [reply pkg/features/kube_features.go:1059]: > alpha is not the thing failing the unit tests, defaulting back to false is what's failing, and means we have to revert the unit test changes as well

Sorry for my mistake, I was still working just now and only briefly looked at it (although it's already 2 AM😩). 

If everyone thinks using the deprecated gate is better, I will agree and will update the KEP with it later.

Comment (liggitt) [reply pkg/features/kube_features.go:1059]: heh, no worries, for gates which are not a new feature going through alpha/beta, we use deprecated

Comment (HirazawaUi) [reply pkg/features/kube_features.go:1059]: > for gates which are not a new feature going through alpha/beta, we use deprecated

I 100% agree with this approach. When this KEP was implemented, we hadn't clearly defined what a deprecated gate was (at least not as clearly as we do now) :-)

**Comment by sftim** [comment]
We're all human, apart from @k8s-ci-robot

File: pkg/features/kube_features.go (hunk 1 above)
Comment (tengqm) [reply pkg/features/kube_features.go:1059]: Are we making a mistake here?
We are deprecating the `kubeproxyVersion` field, not the feature gate.
The feature gate `DisableNodeKubeProxyVersion` goes from alpha to beta, to GA.
The gate is not deprecated. A deprecated gate means it is no longer effective...

Am I understanding this correctly?

Comment (tengqm) [reply pkg/features/kube_features.go:1059]: When we are confident with this change, we mark the gate as "Beta", although the default value is "false". That should be fine. Right?

Comment (HirazawaUi) [reply pkg/features/kube_features.go:1059]: Please refer to: https://github.com/kubernetes/kubernetes/pull/129713#discussion_r1947501872

It explains why this feature gate is different :)

**Pull request merged**